spec/parser_spec.lua

```diff
@@ -643,8 +643,8 @@ describe("Titan parser", function()
 
         assert_program_ast([[
             record List
-                p: {Point}
-                next: List
+                p: {Point};
+                next: List;
```

Comment:
I think this deserves to be in its own test case instead of piggybacking on top of an existing test for recursive type declarations.

Reply:
Agreed, I will add another test cases, please don't merge this yet.